source/features/no-unnecessary-split-diff-view.css

```diff
@@ -1,13 +1,29 @@
-[data-rgh-hide-empty-split-diff-side] {
-	table-layout: auto !important;
-}
-
-[data-rgh-hide-empty-split-diff-side='left'] :is([data-hunk], .blob-expanded) td:nth-child(2),
-[data-rgh-hide-empty-split-diff-side='right'] :is([data-hunk], .blob-expanded) td:nth-child(4) {
+.rgh-no-unnecessary-split-diff-view
+	.js-diff-table:not(:has([data-split-side="left"]:is(.blob-code-addition, .blob-code-deletion)))
+	:is([data-hunk], .blob-expanded)
+	td:nth-child(2),
+.rgh-no-unnecessary-split-diff-view
+	.js-diff-table:not(:has([data-split-side="right"]:is(.blob-code-addition, .blob-code-deletion)))
+	:is([data-hunk], .blob-expanded)
+	td:nth-child(4) {
 	display: none !important;
 	width: 0 !important;
 }
 
-[data-rgh-hide-empty-split-diff-side] :is(.inline-comments, .js-inline-annotations) .empty-cell:not(.blob-num) {
+.rgh-no-unnecessary-split-diff-view
+	.js-diff-table:not(:has([data-split-side="left"]:is(.blob-code-addition, .blob-code-deletion))),
+.rgh-no-unnecessary-split-diff-view
+	.js-diff-table:not(:has([data-split-side="right"]:is(.blob-code-addition, .blob-code-deletion))) {
+	table-layout: auto !important;
+}
+
+.rgh-no-unnecessary-split-diff-view
+	.js-diff-table:not(:has([data-split-side="left"]:is(.blob-code-addition, .blob-code-deletion)))
+	:is(.inline-comments, .js-inline-annotations)
+	.empty-cell:not(.blob-num),
+.rgh-no-unnecessary-split-diff-view
+	.js-diff-table:not(:has([data-split-side="right"]:is(.blob-code-addition, .blob-code-deletion)))
+	:is(.inline-comments, .js-inline-annotations)
```

Comment:
No nesting, no custom selectors available. It would be much more readable 🥲 

There's likely room for improvement though, I just copy-pasted the previous selectors and tweaked it until it worked.